dvc/stage/loader.py

```diff
@@ -24,6 +24,9 @@ def __init__(self, dvcfile, data, lockfile_data=None):
         self.dvcfile = dvcfile
         self.data = data or {}
         self.stages_data = self.data.get("stages", {})
+        self.metrics_data = self.data.get("metrics", [])
+        self.params_data = self.data.get("params", [])
+        self.plots_data = self.data.get("plots", {})
```

Comment:
The whole PR is standing on this 3 lines. It feels weird to use `StageLoader` to get other sections, but this was a quick-and-easy solution.

Reply:
Need to rearchitect here.